pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -3600,6 +3632,47 @@ func pruneRegistrationEntries(tx *gorm.DB, expiresBefore time.Time, logger logru
 	return nil
 }
 
+func createRegistrationEntryEvent(tx *gorm.DB, entryID string) error {
+	if !fflag.IsSet(fflag.FlagEventsBasedCache) {
+		return nil
+	}
+
+	newRegisteredEntryEvent := RegisteredEntryEvent{
+		EntryID: entryID,
+	}
+
+	if err := tx.Create(&newRegisteredEntryEvent).Error; err != nil {
+		return sqlError.Wrap(err)
+	}
+
+	return nil
+}
+
+func listRegistrationEntriesEvents(tx *gorm.DB, req *datastore.ListRegistrationEntriesEventsRequest) (*datastore.ListRegistrationEntriesEventsResponse, error) {
+	var events []RegisteredEntryEvent
```

Comment:
Since this is experimental it must be called only when feature flag is enabled

```suggestion
func listRegistrationEntriesEvents(tx *gorm.DB, req *datastore.ListRegistrationEntriesEventsRequest) (*datastore.ListRegistrationEntriesEventsResponse, error) {
	if !fflag.IsSet(fflag.FlagEventsBasedCache) {
		return &datastore.ListRegistrationEntriesEventsRespons{}, nil
	}
	var events []RegisteredEntryEvent
```